### PR TITLE
Remove some `# type: ignore` from `kornia.feature`

### PR DESCRIPTION
**Pull request by johnnv1**
Related to #1992 -- remove some type ignores and add `kornia.core` for these files

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]

File: kornia/feature/keynet.py [hunk 1]
```diff
@@ -186,8 +186,8 @@ def __init__(
         pretrained: bool = False,
         num_features: int = 2048,
         keynet_conf: Dict = keynet_config['KeyNet_default_config'],
-        ori_module: nn.Module = PassLAF(),
-        aff_module: nn.Module = PassLAF(),
+        ori_module: Module = PassLAF(),
```
Comment (edgarriba) [inline kornia/feature/keynet.py:189]:
```suggestion
        ori_module: Optional[Module] = None,
```
In the code, if none then PassLaf()

File: kornia/feature/keynet.py [hunk 2]
```diff
@@ -160,11 +160,11 @@ def forward(self, x: Tensor) -> Tensor:
             feats_i = self.feature_extractor(x)
             feats_i = F.interpolate(feats_i, size=(shape_im[2], shape_im[3]), mode='bilinear')
             feats.append(feats_i)
-        scores = self.last_conv(torch.cat(feats, dim=1))
+        scores = self.last_conv(concatenate(feats, dim=1))
```
Comment (edgarriba) [inline kornia/feature/keynet.py:163]:
```suggestion
        scores = self.last_conv(concatenate(feats, 1))
```
numpy/jax axis, torch dim

Or maybe time to define our custom concatenate that accept both

File: kornia/feature/laf.py [hunk 3]
```diff
@@ -51,18 +52,18 @@ def get_laf_center(LAF: torch.Tensor) -> torch.Tensor:
...
     return out
 
 
-def get_laf_orientation(LAF: torch.Tensor) -> torch.Tensor:
+def get_laf_orientation(LAF: Tensor) -> Tensor:
     """Return orientation of the LAFs, in degrees.
 
     Args:
-        LAF: (torch.Tensor): tensor [BxNx2x3].
+        LAF: (Tensor): tensor [BxNx2x3].
```
Comment (edgarriba) [inline kornia/feature/laf.py:63]:
```suggestion
        LAF: tensor [BxNx2x3].
```
Types are inferred automatically in docs

File: kornia/feature/laf.py [hunk 4]
```diff
@@ -93,16 +94,14 @@ def set_laf_orientation(LAF: torch.Tensor, angles_degrees: torch.Tensor) -> torc
     """
     KORNIA_CHECK_LAF(LAF)
     B, N = LAF.shape[:2]
-    rotmat: torch.Tensor = angle_to_rotation_matrix(angles_degrees).view(B * N, 2, 2)
-    laf_out: torch.Tensor = torch.cat(
+    rotmat: Tensor = angle_to_rotation_matrix(angles_degrees).view(B * N, 2, 2)
+    laf_out: Tensor = torch.cat(
```
Comment (edgarriba) [inline kornia/feature/laf.py:98]:
```suggestion
    laf_out: Tensor  = concatenate (
```

File: kornia/feature/laf.py [hunk 5]
```diff
@@ -182,7 +181,7 @@ def make_upright(laf: torch.Tensor, eps: float = 1e-9) -> torch.Tensor:
     # matrix to an identity: U, S, V = svd(LAF); LAF_upright = U * S.
     b2a2 = torch.sqrt(laf[..., 0:1, 1:2] ** 2 + laf[..., 0:1, 0:1] ** 2) + eps
     laf1_ell = torch.cat([(b2a2 / det).contiguous(), torch.zeros_like(det)], dim=3)
-    laf2_ell = torch.cat(  # type: ignore
+    laf2_ell = torch.cat(
```
Comment (edgarriba) [inline kornia/feature/laf.py:184]:
```suggestion
    laf2_ell = concatenate (
```

File: kornia/feature/laf.py [hunk 6]
```diff
@@ -269,15 +268,15 @@ def laf_to_boundary_points(LAF: torch.Tensor, n_pts: int = 50) -> torch.Tensor:
         dim=1,
     )
     # Add origin to draw also the orientation
-    pts = torch.cat([torch.tensor([0.0, 0.0, 1.0]).view(1, 3), pts], dim=0).unsqueeze(0).expand(B * N, n_pts, 3)
+    pts = torch.cat([tensor([0.0, 0.0, 1.0]).view(1, 3), pts], dim=0).unsqueeze(0).expand(B * N, n_pts, 3)
```
Comment (edgarriba) [inline kornia/feature/laf.py:271]:
```suggestion
    pts = concatenate([tensor([0.0, 0.0, 1.0]).view(1, 3), pts], dim=0).unsqueeze(0).expand(B * N, n_pts, 3)
```

File: kornia/feature/laf.py [hunk 7]
```diff
@@ -413,7 +412,7 @@ def extract_patches_simple(
     """
     KORNIA_CHECK_LAF(laf)
     if normalize_lafs_before_extraction:
-        nlaf: torch.Tensor = normalize_laf(laf, img)
+        nlaf = normalize_laf(laf, img)
```
Comment (edgarriba) [inline kornia/feature/laf.py:415]: Ideally this should be type after line 413

File: kornia/feature/laf.py [hunk 8]
```diff
@@ -451,7 +447,7 @@ def extract_patches_from_pyramid(
     """
     KORNIA_CHECK_LAF(laf)
     if normalize_lafs_before_extraction:
-        nlaf: torch.Tensor = normalize_laf(laf, img)
+        nlaf: Tensor = normalize_laf(laf, img)
```
Comment (edgarriba) [inline kornia/feature/laf.py:450]: Typed before if else

**Review by edgarriba — COMMENTED** [review]
Some potential improvements towards framework generalization

File: kornia/feature/laf.py (hunk 7 above)
Comment (johnnv1) [reply kornia/feature/laf.py:415]: this explicit type is not even needed here

File: kornia/feature/laf.py (hunk 6 above)
Comment (johnnv1) [reply kornia/feature/laf.py]: Done on [cd80cf1](https://github.com/kornia/kornia/pull/1995/commits/cd80cf17facdf1c2adf6f630b51ee21eb1a1e826)

File: kornia/feature/laf.py (hunk 5 above)
Comment (johnnv1) [reply kornia/feature/laf.py]: Done on [cd80cf1](https://github.com/kornia/kornia/pull/1995/commits/cd80cf17facdf1c2adf6f630b51ee21eb1a1e826)

File: kornia/feature/laf.py (hunk 4 above)
Comment (johnnv1) [reply kornia/feature/laf.py]: Done on [cd80cf1](https://github.com/kornia/kornia/pull/1995/commits/cd80cf17facdf1c2adf6f630b51ee21eb1a1e826)

File: kornia/feature/keynet.py (hunk 1 above)
Comment (johnnv1) [reply kornia/feature/keynet.py]: Done on [cd80cf1](https://github.com/kornia/kornia/pull/1995/commits/cd80cf17facdf1c2adf6f630b51ee21eb1a1e826)
